conda.el

```diff
@@ -263,10 +263,11 @@ Set for the lifetime of the process.")
 
 (defun conda--infer-env-from-buffer ()
   "Search up the project tree for an `environment.yml` defining a conda env."
-  (let ((filename (buffer-file-name)))
-    (when filename
+  (let ((filename (buffer-file-name))
```

Comment:
I think we need the `buffer-file-name` to come first here, to ensure that if it's defined it's chosen first. For example:
```suggestion
  (let ((working-file-or-dir (or (buffer-file-name) default-directory)))
    (when working-file-or-dir
      (conda--get-name-from-env-yml (conda--find-env-yml (f-dirname working-file-or-dir)))
```

Reply:
yes, it makes sense. Sorry that I over extrapolated my use case.